### PR TITLE
Add support for Extra volume and Volume mounts

### PR DESCRIPTION
**Pull request by reshmarajan1730**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have bumped the version number in `/charts/flagsmith/Chart.yaml` in the section `version` or I'm merging to a release branch

## Changes

To fetch external database credentials stored as secrets in the Azure key vault, It's required to mount secrets to the container filesystem.
If DATABASE_URL is not used, Env vars should be used for DB connection

## How did you test this code?

Tested deployment in own environment

### CODE REVIEW

**Comment by dabeeeenster** [comment]
Thanks for the PR! You need a version number bump please!

File: charts/flagsmith/templates/_api_environment.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{{- if .Values.databaseExternal.url }}
```
Comment (plumdog) [inline charts/flagsmith/templates/_api_environment.yaml:1]: I don't think this is right.

`DATABASE_URL` needs to be set, and `databaseExternal.url` is one of the ways it can get set, but not the only way.

Why is this change needed and how does it relate to the extra volumes and volume mounts?

File: charts/flagsmith/templates/deployment-api.yaml [hunk 2]
```diff
@@ -82,6 +82,8 @@ spec:
         ports:
         - containerPort: {{ .Values.service.api.port }}
         env: {{ include (print $.Template.BasePath "/_api_environment.yaml") . | nindent 8 }}
+        extraVolumeMounts:
+          {{- toYaml .Values.extraVolumeMounts | default "" | nindent 12 }}
```
Comment (plumdog) [inline charts/flagsmith/templates/deployment-api.yaml:86]:
```suggestion
        {{- with .Values.api.extraVolumeMounts }}
        volumeMounts: {{- toYaml . | nindent 8 }}
        {{- end }}
```
I don't think `extraVolumeMounts` is a valid key here in `v1/Pod`.

File: charts/flagsmith/templates/deployment-api.yaml [hunk 3]
```diff
@@ -104,3 +106,6 @@ spec:
           timeoutSeconds: {{ .Values.api.readinessProbe.timeoutSeconds }}
         resources:
 {{ toYaml .Values.api.resources | indent 10 }}
+      extraVolumes:
+        {{- toYaml .Values.extraVolumes | default "" | nindent 8 }}     
+
```
Comment (plumdog) [inline charts/flagsmith/templates/deployment-api.yaml:111]:
```suggestion
      {{- with .Values.extraVolumes }}
      volumes: {{- toYaml .Values.api.extraVolumes | nindent 6 }}
      {{- end }}
```

File: charts/flagsmith/templates/secrets-api.yaml [hunk 4]
```diff
@@ -7,7 +7,9 @@ metadata:
     app.kubernetes.io/component: api
 type: Opaque
 data:
+{{- if .Values.databaseExternal.url }}
```
Comment (plumdog) [inline charts/flagsmith/templates/secrets-api.yaml:10]: Similarly to https://github.com/Flagsmith/flagsmith-charts/pull/102/files#r987786252, I don't think this is right.

**Comment by plumdog** [comment]
You should set
```yaml
api:
  extraVolumes: null
  extraVolumeMounts: null
```
in `values.yaml` and note (or link to k8s docs) to explain what values they expect.

Also, as I understand it, from reading https://learn.microsoft.com/en-us/azure/aks/csi-secrets-store-driver#set-an-environment-variable-to-reference-kubernetes-secrets, once the key is set as a volume, you then reference the existing secret. I think this would be supported by setting the following in your values:
```yaml
databaseExternal:
  enabled: true
  urlFromExistingSecret:
    enabled: true
    name: my-secret
    key: key-in-secret
```
I think that should work instead of the `{{- if .Values.databaseExternal.url }}` changes.

**Comment by dabeeeenster** [comment]
Hi @reshmarajan1730 - closing this - please reopen if you have further input - thanks!
